vllm/grpc/grpc_request_manager.py

```diff
@@ -0,0 +1,320 @@
+        elif constraint_field == "regex":
+            structured_outputs = StructuredOutputsParams(regex=proto_params.regex)
+        elif constraint_field == "grammar":
+            structured_outputs = StructuredOutputsParams(grammar=proto_params.grammar)
+        elif constraint_field == "structural_tag":
+            structured_outputs = StructuredOutputsParams(
+                structural_tag=proto_params.structural_tag
+            )
+        elif constraint_field == "json_object":
+            structured_outputs = StructuredOutputsParams(
+                json_object=proto_params.json_object
+            )
+        elif constraint_field == "choice":
+            structured_outputs = StructuredOutputsParams(
+                choice=list(proto_params.choice.choices)
+            )
+
+    # Handle logit_bias
+    logit_bias = None
+    if proto_params.logit_bias:
+        logit_bias = dict(proto_params.logit_bias)
+
+    # Create SamplingParams with detokenize=False and output_kind=DELTA
+    # detokenize=False: KEY OPTIMIZATION that skips detokenization!
+    # output_kind=DELTA: Return only new tokens in each chunk (for streaming)
+    return SamplingParams(
+        temperature=proto_params.temperature if proto_params.temperature > 0 else 1.0,
+        top_p=proto_params.top_p if proto_params.top_p > 0 else 1.0,
+        top_k=proto_params.top_k if proto_params.top_k > 0 else -1,
+        min_p=proto_params.min_p if proto_params.min_p > 0 else 0.0,
+        frequency_penalty=proto_params.frequency_penalty,
+        presence_penalty=proto_params.presence_penalty,
+        repetition_penalty=proto_params.repetition_penalty
+        if proto_params.repetition_penalty > 0
+        else 1.0,
+        max_tokens=proto_params.max_tokens
+        if proto_params.HasField("max_tokens")
+        else None,
+        min_tokens=proto_params.min_tokens if proto_params.min_tokens > 0 else 0,
+        stop=stop,
+        stop_token_ids=stop_token_ids,
+        skip_special_tokens=proto_params.skip_special_tokens,
+        spaces_between_special_tokens=proto_params.spaces_between_special_tokens,
+        ignore_eos=proto_params.ignore_eos,
+        n=proto_params.n if proto_params.n > 0 else 1,
+        logprobs=proto_params.logprobs if proto_params.HasField("logprobs") else None,
+        prompt_logprobs=proto_params.prompt_logprobs
+        if proto_params.HasField("prompt_logprobs")
+        else None,
+        seed=proto_params.seed if proto_params.HasField("seed") else None,
+        include_stop_str_in_output=proto_params.include_stop_str_in_output,
+        logit_bias=logit_bias,
+        truncate_prompt_tokens=proto_params.truncate_prompt_tokens
+        if proto_params.HasField("truncate_prompt_tokens")
+        else None,
+        structured_outputs=structured_outputs,
+        detokenize=False,
+        output_kind=RequestOutputKind.DELTA if stream else RequestOutputKind.CUMULATIVE,
```

Comment:
Actually both `RequestOutputKind.CUMULATIVE` and `RequestOutputKind.DELTA` are streaming, for non-streaming we would want to use `RequestOutputKind.FINAL_ONLY`.